service/hook/gogs/gogs.go

```diff
@@ -86,13 +91,49 @@ func transformCodePushEvent(codePushEvent CodePushEventModel) hookCommon.Transfo
 		}
 	}
 
+	if len(lastCommit.CommitHash) == 0 {
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Missing commit hash"),
+		}
+	}
+
+	if !strings.HasPrefix(pushEvent.Ref, "refs/heads/") {
+		return hookCommon.TransformResultModel{
+			Error: fmt.Errorf("Ref (%s) is not a head ref", pushEvent.Ref),
+		}
+	}
+
+	branch := strings.TrimPrefix(pushEvent.Ref, "refs/heads/")
+
 	return hookCommon.TransformResultModel{
 		TriggerAPIParams: []bitriseapi.TriggerAPIParamsModel{
 			{
 				BuildParams: bitriseapi.BuildParamsModel{
+					Branch:        branch,
 					CommitHash:    lastCommit.CommitHash,
 					CommitMessage: lastCommit.CommitMessage,
-					Branch:        branch,
+				},
+			},
+		},
+	}
+}
+
+func transformCreateEvent(createEvent CreateEventModel) hookCommon.TransformResultModel {
+	// Currently, we only support tag creation, not branches. Even if we wanted branch creation
+	// to trigger a build we would have to wait for https://github.com/go-gitea/gitea/issues/286
+	// to be in both Gogs and Gitea core, and well adopted/distributed.
+	if createEvent.RefType != "tag" {
+		return hookCommon.TransformResultModel{
+			Error:      errors.New("Ignoring branch-create request"),
+			ShouldSkip: true,
+		}
+	}
+
+	return hookCommon.TransformResultModel{
+		TriggerAPIParams: []bitriseapi.TriggerAPIParamsModel{
+			{
+				BuildParams: bitriseapi.BuildParamsModel{
+					Tag: createEvent.Ref,
```

Comment:
Can you please add the commit hash and commit message params as well? Not sure if gitea/gogs supports these, but where the system supports it we send those details too - see e.g. the github one: https://github.com/bitrise-io/bitrise-webhooks/blob/365edb90077b9ece88a0319809301f311b92c475/service/hook/github/github.go#L144 or the Bitbucket.org one https://github.com/bitrise-io/bitrise-webhooks/blob/365edb90077b9ece88a0319809301f311b92c475/service/hook/bitbucketv2/bitbucketv2.go#L171